src/hypertable_restrict_info.c

```diff
@@ -97,7 +97,7 @@ dimension_restrict_info_open_add(DimensionRestrictInfoOpen *dri, StrategyNumber
 
 	foreach(item, dimValues->values)
 	{
-		Datum		datum = PointerGetDatum(lfirst(item));
+		Datum		datum = dimension_transform_value(dri->base.dimension, PointerGetDatum(lfirst(item)));
```

Comment:
@cevian note this change here, using the transformed value (partitioning func applied) as opposed to the plain value. I think this is correct for this optimization, but would like you to double check.

Reply:
This looks fine. But the line below does not. Namely, dimValues->type is the value of the `time` column not the type of the /transformed/ value.